scripts/install-latest-ci.sh

```diff
@@ -1,101 +1,148 @@
+    # wslpath "$(cmd.exe /c "echo %$1%" 2>/dev/null)" | tr -d "\r"
+    # Let's see if powershell.exe avoids this issue
+    wslpath "$(powershell.exe -Command "Write-Output \${Env:$1}")" | tr -d "\r"
+}
 
-unzip -o "$TMPDIR/$FILENAME" "$ZIP" -d "$TMPDIR"
+install_application() {
+    local archive
+    if [[ "$(uname -s)" == "Darwin" ]]; then
+        ARCH=x86_64
+        if [ "$(uname -m)" = "arm64" ]; then
+            ARCH=aarch64
+        fi
+        archive="${TMPDIR:-/tmp}/Rancher Desktop-mac.$ARCH.zip"
+    elif [[ "$(uname -r)" =~ "WSL2" ]]; then
+        archive="${TMPDIR:-/tmp}/Rancher Desktop-win.zip"
+    else
+        archive="${TMPDIR:-/tmp}/Rancher Desktop-linux.zip"
+    fi
+    download_artifact "$archive"
+
+    # Artifacts are zipped, so extract inner ZIP file from outer wrapper.
+    # The outer ZIP has a predictable name like "Rancher Desktop-mac.x86_64.zip"
+    # but the inner one has a version string: "Rancher Desktop-1.7.0-1061-g91ab3831-mac.zip"
+    # Run unzip in "zipinfo" mode, which can print just the file name.
+    local ZIP
+    ZIP="$(unzip -Z -1 "$archive" | head -n1)"
+    if [ -z "$ZIP" ]; then
+        echo "Cannot find inner archive in $(basename "$archive")"
+        exit 1
+    fi
+    local zip_full="${TMPDIR:-/tmp}/$ZIP"
+
+    unzip -o "$archive" "$ZIP" -d "${TMPDIR:-/tmp}"
+    local DEST
+
+    if [[ "$(uname -s)" == "Darwin" ]]; then
+        # Extract from inner archive into /Applications
+        DEST="/Applications"
+        if [ "$RD_LOCATION" = "user" ]; then
+            DEST="$HOME/$DEST"
+        fi
+
+        APP="Rancher Desktop.app"
+        rm -rf "${DEST:?}/$APP"
+        unzip -o "$zip_full" "$APP/*" -d "$DEST" >/dev/null
+    elif [[ "$(uname -r)" =~ "WSL2" ]]; then
+        LOCALAPPDATA="$(wslpath_from_win32_env LOCALAPPDATA)"
+        PROGRAMFILES="$(wslpath_from_win32_env ProgramFiles)"
+        DEST="$PROGRAMFILES"
+        if [ "$RD_LOCATION" = "user" ]; then
+            DEST="$LOCALAPPDATA/Programs"
+        fi
+        APP="Rancher Desktop"
+        rm -rf "${DEST:?}/$APP"
+        unzip -o "$zip_full" "$APP/*" -d "$DEST" >/dev/null
+    else
+        # Linux doesn't support per-user installs
+        DEST="/opt/rancher-desktop"
+        sudo rm -rf "${DEST:?}"
+        sudo unzip -o "$zip_full" -d "$DEST" >/dev/null
+    fi
+}
 
-# Extract from inner archive into /Applications
-DEST="/Applications"
-if [ "$RD_LOCATION" = "user" ]; then
-    DEST="$HOME/$DEST"
-fi
+download_bats() {
+    pushd "${TMPDIR:-/tmp}"
+    download_artifact "bats.tar.gz"
+
+    # GitHub always wraps the artifact in a zip file, so the downloaded file
+    # actually has an incorrect name; extract it in place.
+    mv bats.tar.gz bats.tar.gz.zip
+    unzip -o bats.tar.gz.zip bats.tar.gz
+
+    # Unpack bats into $BATS_DIR
+    rm -rf "$BATS_DIR"
+    mkdir "$BATS_DIR"
+    tar xfz "bats.tar.gz" -C "$BATS_DIR"
+    popd
+}
+
+API_ARGS="exclude_pull_requests=true"
```

Comment:
Why is this one needed? I may remember the branch name for my PR, so I may try to download by name instead of PR#, which I may not remember.

The was originally the only way to download builds; I just added the `PR` setting as a shortcut when I wanted to install builds from other user's PRs.

Is there a risk of selecting the wrong workflow run without this parameter?

Reply:
This isn't really needed, no. It's just there to reduce the number of possible things to look at; I'll drop it.

If you know the branch name, you can (and should) just use that from your fork instead (override `$OWNER`); you still wouldn't need to use artifacts from the upstream repo.